iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -49,458 +50,528 @@ namespace mlir {
-        b.createOrFold<memref::DimOp>(loc, inputBuffer, dim));
-  }
-  if (failed(createAliasingBufferOrAllocationForResult(
-          b, loc, allocationFn, dest, inputBuffer, op.getResult(),
-          allocationDynamicSizes, bvm))) {
+static LogicalResult analyseOperations(FuncOp funcOp, BufferizationPlan &plan) {
+  auto bufferMappingFn = [&](Operation *op) -> WalkResult {
+    return TypeSwitch<Operation *, LogicalResult>(op)
+        .Case<ConstantOp>([&](ConstantOp constantOp) {
+          return analyseConstantOp(constantOp, plan);
+        })
+        .Case<IREE::Flow::DispatchTensorLoadOp>(
+            [&](IREE::Flow::DispatchTensorLoadOp loadOp) {
+              return analyseLoadOp(loadOp.source(), loadOp.result(), plan);
+            })
+        .Case<IREE::Flow::DispatchTensorStoreOp>(
+            [&](IREE::Flow::DispatchTensorStoreOp storeOp) {
+              return analyseInterfaceStoreTensorOp(storeOp, plan);
+            })
+        .Case<IREE::Flow::DispatchTieShapeOp>(
+            [&](IREE::Flow::DispatchTieShapeOp tieShapeOp) {
+              return analyseSingleOperandResultOp(tieShapeOp.operand(),
+                                                  tieShapeOp.result(), plan);
+            })
+        .Case<IREE::HAL::InterfaceBindingSubspanOp>(
+            [&](IREE::HAL::InterfaceBindingSubspanOp subspanOp) {
+              return analyseInterfaceBindingSubspanOp(subspanOp, plan);
+            })
+        .Case<linalg::LinalgOp>([&](linalg::LinalgOp linalgOp) {
+          return analyseLinalgOps(linalgOp, plan);
+        })
+        .Case<linalg::TensorReshapeOp>(
+            [&](linalg::TensorReshapeOp tensorReshapeOp) {
+              return analyseSingleOperandResultOp(
+                  tensorReshapeOp.src(), tensorReshapeOp.result(), plan);
+            })
+        .Case<SubTensorOp>([&](SubTensorOp subTensorOp) {
+          return analyseSingleOperandResultOp(subTensorOp.source(),
+                                              subTensorOp.result(), plan);
+        })
+        .Case<SubTensorInsertOp>([&](SubTensorInsertOp subTensorInsertOp) {
+          return analyseDestructiveUpdateOp(
+              subTensorInsertOp, subTensorInsertOp.source(),
+              subTensorInsertOp.dest(), subTensorInsertOp.result(), plan);
+        })
+        .Case<tensor::CastOp>([&](tensor::CastOp castOp) {
+          return analyseSingleOperandResultOp(castOp.source(), castOp.dest(),
+                                              plan);
+        })
+        .Case<vector::TransferReadOp>(
+            [&](vector::TransferReadOp transferReadOp) {
+              return analyseLoadOp(transferReadOp.source(),
+                                   transferReadOp.vector(), plan);
+            })
+        .Case<vector::TransferWriteOp>(
+            [&](vector::TransferWriteOp transferWriteOp) {
+              return analyseDestructiveUpdateOp(
+                  transferWriteOp, transferWriteOp.vector(),
```

Comment:
same comment, why do we care about the vector?

Reply:
Fixed this too.